aws-api/src/test/java/com/amplifyframework/api/aws/SubscriptionAuthorizerTest.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.api.aws;
+
+import com.amplifyframework.api.ApiException;
+import com.amplifyframework.api.aws.sigv4.CognitoUserPoolsAuthProvider;
+import com.amplifyframework.testutils.random.RandomString;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Test that Subscription authorizer can correctly generate appropriate
+ * headers for different types of authentication methods.
+ */
+@RunWith(RobolectricTestRunner.class)
+public final class SubscriptionAuthorizerTest {
+    private String authenticationSecret;
+    private ApiAuthProviders apiAuthProviders;
+
+    /**
+     * Construct fake auth providers to override default behaviors for
+     * subscription authorizer.
+     */
+    @Before
+    public void setup() {
+        authenticationSecret = RandomString.string();
+        apiAuthProviders = ApiAuthProviders.builder()
```

Comment:
I've never liked this `ApiAuthProviders` thing, it has poor encapsulation, and muddles together several different technologies without a consistent behavioral contract.

Reply:
Yea it was a poor implementation on my part back when I was working on it for mutate/query, but it's a legacy that should stay for until after GA...